src/psykal/psy.f90

```diff
@@ -2,48 +2,78 @@ module psy_mod
 !
 implicit none
 !
+private
+public :: set_bounds
+public :: zero_top_layer, zero_bottom_layer, multiply_bottom_layer
+public :: zind_psy, zwxy_psy, zslpxy_psy, zslpxy_update_psy, zwxy2_psy, &
+          mydomain_update_psy, zwx_psy, zslpx_psy, zslpx_update_psy, &
+          zwx2_psy, mydomain_psy
+!
+integer :: jpk, jpj, jpi
+!
 contains
   !
-  subroutine zero_layer(field,jpj,jpi)
+  subroutine set_bounds(jpk_in,jpj_in,jpi_in)
+    !
+    integer, intent(in) :: jpk_in,jpj_in,jpi_in
+    !
+    jpk=jpk_in
+    jpj=jpj_in
+    jpi=jpi_in
+    !
+  end subroutine set_bounds
+  !
+  subroutine zero_top_layer(field)
     !
-    real*8, intent(out) :: field(:,:)
-    integer, intent(in) :: jpj,jpi
+    real*8, intent(inout) :: field(jpi,jpj,jpk)
     ! local variables
     integer :: jj,ji
     !
     DO jj=1,jpj
       DO ji=1,jpi
-        field(ji,jj) = 0.e0
+        field(ji,jj,jpk) = 0.e0
```

Comment:
Oh, and in NEMO, the convention is that jk=1 is the top layer (surface) while jk=jpk is the bottom layer. This is a model-dependent choice (although I think it is a common one). Perhaps we need meta-data for this?
Therefore, this routine should be named `zero_bottom_layer`.